diff --git a/cmake/partition_manager.cmake b/cmake/partition_manager.cmake
index a45650a2..292dd9c3 100644
--- a/cmake/partition_manager.cmake
+++ b/cmake/partition_manager.cmake
@@ -76,18 +76,6 @@ endif()
 
 if (NOT static_configuration AND
   (CONFIG_BOOTLOADER_MCUBOOT OR CONFIG_SECURE_BOOT))
-      message(WARNING "
-        ---------------------------------------------------------------------
-        --- WARNING: Using a bootloader without pm_static.yml.            ---
-        --- There are cases where a deployed product can consist of       ---
-        --- multiple images, and only a subset of these images can be     ---
-        --- upgraded through a firmware update mechanism. In such cases,  ---
-        --- the upgradable images must have partitions that are static    ---
-        --- and are matching the partition map used by the bootloader     ---
-        --- programmed onto the device.                                   ---
-        ---------------------------------------------------------------------
-        \n"
-      )
 endif()
 
 
diff --git a/modules/mcuboot/CMakeLists.txt b/modules/mcuboot/CMakeLists.txt
index 5ec2643d..70c706d5 100644
--- a/modules/mcuboot/CMakeLists.txt
+++ b/modules/mcuboot/CMakeLists.txt
@@ -379,13 +379,6 @@ if(CONFIG_BOOTLOADER_MCUBOOT)
   if(CONFIG_SIGN_IMAGES)
     # Set default key
     if (NOT DEFINED mcuboot_key_file)
-      message(WARNING "
-        ---------------------------------------------------------
-        --- WARNING: Using default MCUBoot key, it should not ---
-        --- be used for production.                           ---
-        ---------------------------------------------------------
-        \n"
-      )
       set(mcuboot_key_file ${ZEPHYR_MCUBOOT_MODULE_DIR}/${CONFIG_BOOT_SIGNATURE_KEY_FILE})
     endif()
 
