From e7e191c9040a91ac0b6288cf88e1769c1c97c01b Mon Sep 17 00:00:00 2001
From: Benedek Kupper <kupper.benedek@gmail.com>
Date: Thu, 19 Jun 2025 04:38:29 +0200
Subject: [PATCH] kconfig: don't use config header from drivers

---
 Kconfig | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Kconfig b/Kconfig
index a2d2253..ca3963a 100644
--- a/Kconfig
+++ b/Kconfig
@@ -49,7 +49,7 @@ if MCUX_COMPONENT_middleware.usb.common_header
                 MCUX_HW_IP_DriverType_USB_HOST_DEVICE_KHCI || \
                 MCUX_HW_IP_DriverType_USB_DEVICE_KHCI
             select MCUX_COMPONENT_middleware.usb.device.controller.driver
-            select MCUX_COMPONENT_middleware.usb.device.khci_config_header
+            #select MCUX_COMPONENT_middleware.usb.device.khci_config_header
 
         config MCUX_COMPONENT_middleware.usb.device.khci_config_header
             bool
@@ -60,7 +60,7 @@ if MCUX_COMPONENT_middleware.usb.common_header
             default y
             depends on MCUX_HW_IP_DriverType_USB_HOST_DEVICE_EHCI
             select MCUX_COMPONENT_middleware.usb.device.controller.driver
-            select MCUX_COMPONENT_middleware.usb.device.ehci_config_header
+            #select MCUX_COMPONENT_middleware.usb.device.ehci_config_header
             select MCUX_COMPONENT_middleware.usb.phy if MCUX_HW_IP_DriverType_USB_PHY
             select MCUX_COMPONENT_driver.memory if (MCUX_HW_DEVICE_RW610 || MCUX_HW_DEVICE_RW612)
 
@@ -73,7 +73,7 @@ if MCUX_COMPONENT_middleware.usb.common_header
             default y
             depends on MCUX_HW_IP_DriverType_USB_DEVICE_IP3511FS
             select MCUX_COMPONENT_middleware.usb.device.controller.driver
-            select MCUX_COMPONENT_middleware.usb.device.ip3511fs_config_header
+            #select MCUX_COMPONENT_middleware.usb.device.ip3511fs_config_header
 
         config MCUX_COMPONENT_middleware.usb.device.ip3511fs_config_header
             bool
@@ -84,7 +84,7 @@ if MCUX_COMPONENT_middleware.usb.common_header
             default y
             depends on MCUX_HW_IP_DriverType_USB_DEVICE_IP3511HS
             select MCUX_COMPONENT_middleware.usb.device.controller.driver
-            select MCUX_COMPONENT_middleware.usb.device.ip3511hs_config_header
+            #select MCUX_COMPONENT_middleware.usb.device.ip3511hs_config_header
             select MCUX_COMPONENT_middleware.usb.phy if MCUX_HW_IP_DriverType_USB_PHY
             select MCUX_COMPONENT_driver.common if !MCUX_HW_IP_DriverType_USB_PHY
 
-- 
2.44.0

