From 9216faab94006e1f77fb64a3a4b1d73d711a41e6 Mon Sep 17 00:00:00 2001
From: Benedek Kupper <kupper.benedek@gmail.com>
Date: Mon, 28 Oct 2024 21:58:36 +0100
Subject: [PATCH] boards: link ugl board directory as in-tree

---
 boards/ugl | 1 +
 1 file changed, 1 insertion(+)
 create mode 120000 boards/ugl

diff --git a/boards/ugl b/boards/ugl
new file mode 120000
index 00000000000..a497eea33b5
--- /dev/null
+++ b/boards/ugl
@@ -0,0 +1 @@
+../../firmware-uhk80/boards/ugl
\ No newline at end of file
-- 
2.44.0

