From 67e38cff8129e51203dbb79011d254a211c41840 Mon Sep 17 00:00:00 2001
From: Benedek Kupper <kupper.benedek@gmail.com>
Date: Wed, 13 Aug 2025 00:48:37 +0200
Subject: [PATCH] west sysbuild allows using application cmake presets

---
 scripts/west_commands/build.py | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/scripts/west_commands/build.py b/scripts/west_commands/build.py
index 260945483cf..c34e23b30ad 100644
--- a/scripts/west_commands/build.py
+++ b/scripts/west_commands/build.py
@@ -601,7 +601,26 @@ class Build(Forceable):
             # If no option is set, then enable sysbuild globally
             config_sysbuild = True
 
+        # sysbuild has its own cmake project directory, so we need to
+        # copy the application's CMakePresets.json file and replace the paths
+        sysbuild_presets = SYSBUILD_PROJ_DIR / 'CMakePresets.json'
+        source_presets = pathlib.Path(self.source_dir) / 'CMakePresets.json'
+
         if self.args.sysbuild or (config_sysbuild and not self.args.no_sysbuild):
+            try:
+                if source_presets.exists():
+                    with open(source_presets, 'r') as f:
+                        content = f.read()
+                    content = (content
+                        .replace("${sourceDir}", self.source_dir)
+                        .replace("${sourceParentDir}", str(pathlib.Path(self.source_dir).parent))
+                        .replace("${sourceDirName}", str(pathlib.Path(self.source_dir).name))
+                    )
+                    with open(sysbuild_presets, 'w') as f:
+                        f.write(content)
+            except Exception as e:
+                log.wrn(f'Failed to adopt CMakePresets.json to sysbuild: {e}')
+
             cmake_opts.extend(['-S{}'.format(SYSBUILD_PROJ_DIR),
                                '-DAPP_DIR:PATH={}'.format(self.source_dir)])
         else:
@@ -622,6 +641,15 @@ class Build(Forceable):
             final_cmake_args.extend(cmake_opts)
         run_cmake(final_cmake_args, dry_run=self.args.dry_run)
 
+        # clean up the sysbuild CMakePresets.json file after each run
+        if sysbuild_presets.exists():
+            try:
+                sysbuild_presets.unlink()
+            except FileNotFoundError:
+                pass
+            except Exception as e:
+                log.wrn(f'Failed to remove sysbuild CMakePresets.json: {e}')
+
     def _run_pristine(self):
         _banner('making build dir {} pristine'.format(self.build_dir))
         if not is_zephyr_build(self.build_dir):
-- 
2.43.0

